lib/OpenQA/Setting.pm

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2017-2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Setting;
+
+use warnings;
+use strict;
+
+sub new {
+    my ($class, %args) = @_;
+
+    my $self = {%args};
+    bless $self, $class;
+    return $self;
+}
+
+#replace %NAME% in values with $self->{NAME}
+sub replace_setting {
```

Comment:
Since the class is already called `Setting(s)`, I'd say we can just use a verb here, and perhaps call this `update`. This would go nicely with the `update` method in the DBX classes me thinks.

Reply:
I don't like calling this update - it is not very specific. I would call it `expand_placeholders`.

Reply:
I like `expand_placeholders`.

Reply:
fixed.